From 8691defb9cda5e73755d46c1063e43c5aec1ea8d Mon Sep 17 00:00:00 2001
From: Octavian-Tiberiu Sirbu <octavian-tiberiu.sirbu@nxp.com>
Date: Tue, 26 Aug 2025 16:41:15 +0300
Subject: [PATCH] Added patch for UART delay reduction

Signed-off-by: Octavian-Tiberiu Sirbu <octavian-tiberiu.sirbu@nxp.com>
Upstream-Status: Pending
---
 drivers/tty/serial/fsl_lpuart.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/tty/serial/fsl_lpuart.c b/drivers/tty/serial/fsl_lpuart.c
index f36d50fe056f..eae23fea125d 100644
--- a/drivers/tty/serial/fsl_lpuart.c
+++ b/drivers/tty/serial/fsl_lpuart.c
@@ -1699,7 +1699,7 @@ static void lpuart32_setup_watermark_enable(struct lpuart_port *sport)
 
 	ctrl = lpuart32_read(&sport->port, UARTCTRL);
 	ctrl |= UARTCTRL_RE | UARTCTRL_TE;
-	ctrl |= FIELD_PREP(UARTCTRL_IDLECFG, 0x7);
+	ctrl |= FIELD_PREP(UARTCTRL_IDLECFG, 0x3);
 	lpuart32_write(&sport->port, ctrl, UARTCTRL);
 }
 
-- 
2.43.0

