diff --git a/adapters/x509_openssl.c b/adapters/x509_openssl.c
index cf470589..1dccc920 100644
--- a/adapters/x509_openssl.c
+++ b/adapters/x509_openssl.c
@@ -13,6 +13,13 @@
 #include "openssl/err.h"
 #include "openssl/engine.h"
 
+#include <openssl/store.h>
+#include <openssl/provider.h>
+
+#define OPENSSL30_SE05X_CNF_FILE "/etc/ssl/openssl30_sss_se050.cnf"
+
+#define SE050 1
+
 #ifdef __APPLE__
     #ifndef EVP_PKEY_id
         #define EVP_PKEY_id(evp_key) evp_key->type
@@ -116,6 +123,7 @@ static int load_certificate_chain(SSL_CTX* ssl_ctx, const char* certificate)
     return result;
 }
 
+
 static int load_ecc_key(SSL_CTX* ssl_ctx, EVP_PKEY* evp_key)
 {
     int result;
@@ -164,6 +172,57 @@ static int load_key_RSA(SSL_CTX* ssl_ctx, EVP_PKEY* evp_key)
     return result;
 }
 
+int load_key_from_se050(SSL_CTX* ssl_ctx, const char* x509privatekey)
+{
+        EVP_PKEY *p_client_pkey = NULL;
+        OSSL_LIB_CTX *ossl_ctx = NULL;
+        ossl_ctx =  OSSL_LIB_CTX_new();
+        int result = 0;
+        if (ossl_ctx == NULL) {
+                (void)printf("Error in OSSL_LIB_CTX_new \n");
+        }
+
+        fprintf(stderr, "Load openssl 3.0 config file.\n");
+
+        if (OSSL_LIB_CTX_load_config(ossl_ctx, OPENSSL30_SE05X_CNF_FILE) != 1) {
+                LogError("Error in loading openssl 3.0 config file \n");
+        }
+
+        OSSL_STORE_CTX *p_client_pkey_ctx = NULL;
+        OSSL_STORE_INFO *p_client_pkey_info = NULL;
+
+        p_client_pkey_ctx = OSSL_STORE_open_ex(x509privatekey, ossl_ctx, NULL, NULL, NULL, NULL, NULL, NULL);
+        if (p_client_pkey_ctx == NULL){
+                LogError("Error in OSSL_STORE_open for %s. \n", x509privatekey);
+         }
+
+         p_client_pkey_info = OSSL_STORE_load(p_client_pkey_ctx);
+         if (p_client_pkey_info == NULL){
+                 LogError("Error in OSSL_STORE_load. \n");
+                 OSSL_STORE_close(p_client_pkey_ctx);
+                        }
+
+         p_client_pkey       = OSSL_STORE_INFO_get1_PKEY(p_client_pkey_info);
+         if (p_client_pkey == NULL){
+                 LogError("Error in OSSL_STORE_INFO_get1_PKEY. \n");
+                 OSSL_STORE_close(p_client_pkey_ctx);
+                 OSSL_STORE_INFO_free(p_client_pkey_info);
+         }
+
+         OSSL_STORE_close(p_client_pkey_ctx);
+         OSSL_STORE_INFO_free(p_client_pkey_info);
+         if (load_ecc_key(ssl_ctx, p_client_pkey) != 0)
+         {
+                 LogError("failure loading ECC private key cert");
+                 result = MU_FAILURE;
+         }
+         else
+         {
+                 result = 0;
+         }
+         return result;
+}
+
 int x509_openssl_add_pem_file_key(SSL_CTX* ssl_ctx, const char* x509privatekey)
 {
     int result;
@@ -295,7 +354,12 @@ int x509_openssl_add_credentials(
         // Configure private key.
         if (x509privatekeytype == KEY_TYPE_DEFAULT)
         {
-            result = x509_openssl_add_pem_file_key(ssl_ctx, x509privatekey);
+#if SE050
+            result = load_key_from_se050(ssl_ctx, x509privatekey);
+#else
+            //result = x509_openssl_add_pem_file_key(ssl_ctx, x509privatekey);
+            result = load_ecc_key(ssl_ctx, (EVP_PKEY *)x509privatekey);
+#endif
         }
         else if (x509privatekeytype == KEY_TYPE_ENGINE)
         {
